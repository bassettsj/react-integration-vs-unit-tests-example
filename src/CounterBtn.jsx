import React from "react"

export default function CounterBtn({ onPress, children }) {
  const onClick = e => {
    e.preventDefault()
    onPress()
  }
  return (
    <button type="button" onClick={onClick}>
      {children}
    </button>
  )
}
