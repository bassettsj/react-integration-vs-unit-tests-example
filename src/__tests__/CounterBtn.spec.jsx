import { shallow } from "enzyme"
import React from "react"
import CounterBtn from "../CounterBtn"

describe("<CounterBth />", () => {
  it("should should render", () => {
    expect(shallow(<CounterBtn />)).toMatchSnapshot()
  })

  it("should call on press when clicked", () => {
    const onPress = jest.fn()
    shallow(<CounterBtn onPress={onPress} />).simulate("click", {
      preventDefault() {}
    })
    expect(onPress).toHaveBeenCalled()
  })
})
