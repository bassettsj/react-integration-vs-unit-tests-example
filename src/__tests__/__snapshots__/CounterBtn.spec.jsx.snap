// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterBth /> should should render 1`] = `ShallowWrapper {}`;
